apps/opik-backend/src/test/java/com/comet/opik/api/resources/v1/priv/LlmProviderApiKeyResourceTest.java

```diff
@@ -159,56 +154,37 @@ void updateProviderFail() {
         mockTargetWorkspace(apiKey, workspaceName, workspaceId);
 
         // for non-existing id
-        updateProviderApiKey(UUID.randomUUID(), providerApiKey, apiKey, workspaceName, 404);
+        llmProviderApiKeyResourceClient.updateProviderApiKey(UUID.randomUUID(), providerApiKey, apiKey, workspaceName,
+                404);
     }
 
-    private UUID createProviderApiKey(LlmProvider provider, String providerApiKey, String apiKey, String workspaceName,
-            int expectedStatus) {
-        try (var actualResponse = client.target(URL_TEMPLATE.formatted(baseURI))
-                .request()
-                .accept(MediaType.APPLICATION_JSON_TYPE)
-                .header(HttpHeaders.AUTHORIZATION, apiKey)
-                .header(WORKSPACE_HEADER, workspaceName)
-                .post(Entity.json(ProviderApiKey.builder().provider(provider).apiKey(providerApiKey).build()))) {
-
-            assertThat(actualResponse.getStatusInfo().getStatusCode()).isEqualTo(expectedStatus);
-            if (expectedStatus == 201) {
-                return TestUtils.getIdFromLocation(actualResponse.getLocation());
-            }
-
-            return null;
-        }
-    }
+    @Test
+    @DisplayName("Create and get provider Api Keys List")
+    void createAndGetProviderApiKeyList() {
+
+        String workspaceName = UUID.randomUUID().toString();
+        String apiKey = UUID.randomUUID().toString();
+        String workspaceId = UUID.randomUUID().toString();
+        String providerApiKey = factory.manufacturePojo(String.class);
+
+        mockTargetWorkspace(apiKey, workspaceName, workspaceId);
+
+        // No LLM Provider api keys, expect empty response
+        var actualProviderApiKeyPage = llmProviderApiKeyResourceClient.getAll(workspaceName, apiKey);
+        assertPage(actualProviderApiKeyPage, List.of());
+
+        // Create LLM Provider api key
+        var expectedProviderApiKey = llmProviderApiKeyResourceClient.createProviderApiKey(providerApiKey, apiKey,
```

Comment:
I think that in the case of list logic, it's worth to create several api keys

Reply:
It's impossible:) Requirements is that one workspace can have only 1 key per provider. And now we are going to support only 1 provider.